modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/sequence/steps/StepConfigTable.scala

```diff
@@ -25,7 +47,9 @@ object StepConfigTable {
     val rowCount: Int = settingsList.size
 
     def rowGetter(idx: Int): SettingsRow =
-      settingsList.lift(idx).fold(SettingsRow.Zero)(Function.tupled(SettingsRow.apply))
+      settingsList
+        .lift(idx)
+        .fold(SettingsRow.Zero)(Function.tupled(SettingsRow.apply))
```

Comment:
`SettingsRow` is mutable … does it really make sense to have a global mutable `Zero` value?

Reply:
This is a good question. `SettingsRow` has vars inside as it sits at the border with `js`. it can't be a `val` an interoperate properly. However, the way we use it is effectively immutable.
Yes the compiler cannot prevent in this case somebody changing `Zero`. I suppose it could be constructed at the call site if you think that's better

Reply:
Oh weird, everything has to be a var? Crazy. I think I would feel better if there were a factory method that returned a new empty instance.

Reply:
you can make a `js trait` with `vals` but it could be useless as you cannot have a constructor nor create new instances as `new Trait {`.

I'll change the Zero though